lib/lita/adapters/slack/message_handler.rb

```diff
@@ -165,19 +170,30 @@ def handle_hello
 
         def handle_message
           return unless supported_subtype?
-          return if data["user"] == 'USLACKBOT'
-
-          user = User.find_by_id(data["user"]) || User.create(data["user"])
-
-          return if from_self?(user)
+          return if data["user"] == 'USLACKBOT'        
 
           case data["subtype"]
           when "message_deleted"
-            robot.trigger(:message_deleted, data)
+            user_from_data = data['previous_message']['user']
+            user = User.find_by_id(user_from_data) || User.create(user_from_data)
+            return if from_self?(user)
+            
+            message = create_message(user, previous_message=data['previous_message'])
+            response = Response.new(message, nil) #use nil as the pattern bots need to make their own assessment
+            robot.trigger(:message_deleted, response)
           when "message_changed"
-            robot.trigger(:message_changed, data)
+            user_from_data = data['previous_message']['user']
+            user = User.find_by_id(user_from_data) || User.create(user_from_data)
+            return if from_self?(user)
+
+            message = create_message(user, previous_message=data['previous_message'])
+            response = Response.new(message, nil) #use nil as the pattern bots need to make their own assessment
```

Comment:
This seems to be identical to 177-182. Since it's a fairly robust bit of logic, can this be abstracted out in some fashion?